### DISCUSSION TITLE
Add a view editor for easily naming foreign keys...

### DISCUSSION THREAD

**Discussion by sandbox-apps** [supabase/supabase#16918] [Contribute to Supabase]
# Feature request
I have stumbled upon creating complex join in supabase e.g. for composite key especially if this is hierarchically created.
Also if we can see here in the discussion https://github.com/supabase/supabase/discussions/2469
by default the naming will be long for joins thus making it error prone and not maintainable.

Solution: Renaming the foreign key constraints
New Problem: Development setup overhead which is writing another sql to rename the constraint that depends on how many joins the table will have.

**Comment by MildTomato** [comment]
@joshenlim this may have been fixed now. can you check?
